src/components/Header/DarkModeContext.js

```diff
@@ -0,0 +1,17 @@
+import React, { createContext } from "react";
+
+export const DarkModeTheme = createContext();
+
+
+
+function DarkModeProvider(props) {
+    const [darkmode, setDarkmode] = React.useState(false)
+    return (
+        <DarkModeTheme.Provider value={{ darkmode, setDarkmode }}>
+            {props.children}
+        </DarkModeTheme.Provider>
+    )
+}
+
+export const useTheme = () => React.useContext(DarkModeTheme)
```

Comment:
make a hooks folder inside src directory and pull all custom hooks under one file there